cmd/ig/utils/flags.go

```diff
@@ -90,6 +97,10 @@ func AddCommonFlags(command *cobra.Command, commonFlags *CommonFlags) {
 			})
 		}
 
+		if commonFlags.ContainerdNamespace != constants.K8sContainerdNamespace {
+			os.Setenv(namespaces.NamespaceEnvVar, commonFlags.ContainerdNamespace)
+		}
```

Comment:
TBH, I'm unsure about passing this information in an env variable. IMO, it should go through the runtime config like the socket path. Did you explore that possibility?

Reply:
> Did you explore that possibility?

I see the point in this, I felt [RuntimeConfig](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/container-utils/containerutils.go#L45) was bit too generic for including a namespace field since it is only valid for containerd. What do you think about following?

```
type ContainerdConfig struct {
	Namespace string
}

type RuntimeConfig struct {
	Name             types.RuntimeName
	SocketPath       string
	ContainerdConfig *ContainerdConfig
}
```

Reply:
I think it might work. I wouldn't call it `ContainerdConfig` but something more generic like `ExtraConfig` or similar; naming is always hard. In any case, I wouldn't use `Containerd` in it. WDYT?

Reply:
I experimented with the approach and it is looking like [this](https://github.com/inspektor-gadget/inspektor-gadget/commit/c686936caebba32ec66426b8749d52d7bfc51957). What do you say?

Reply:
Just one comment, for the rest it looks good to me. I would like to have an opinion of @eiffel-fl who had already approved the previous approach.